File: Modulo2-ObjectiveC-Lorena/Exercicio20.m
//
//  main.m
//  Exercicio20-C
//
//  Created by Lorena Rodrigues Bruno on 06/09/2018.
//  Copyright © 2018 Lorena Rodrigues Bruno. All rights reserved.
//

#import <Foundation/Foundation.h>

void figura1() {
    int i, j;
    for(i=0; i<8; i++) {
        for(j=0; j<8; j++) {
            if (j <= i) {
                printf("*");
            } else {
                printf(" ");
            }
        }
        printf("\n");
    }
}

void figura2() {
    int i, j;
    for(i=8; i>0; i--) {
        for(j=0; j<8; j++) {
            if (j < i) {
                printf("*");
            } else {
                printf(" ");
            }
        }
        printf("\n");
    }
}

void figura3() {
    int i, j;
    for(i=0; i<8; i++) {
        for(j=0; j<8; j++) {
            if (j >= i) {
                printf("*");
            } else {
                printf(" ");
            }
        }
        printf("\n");
    }
}

void figura4() {
    int i, j;
    for(i=8; i>0; i--) {
        for(j=0; j<=8; j++) {
            if (j >= i) {
                printf("*");
            } else {
                printf(" ");
            }
        }
        printf("\n");
    }
}

int main(int argc, const char * argv[]) {
    @autoreleasepool {
        figura1();
        printf("\n");
        figura2();
        printf("\n");
        figura3();
        printf("\n");
        figura4();
        printf("\n");

    }
    return 0;
}

File: Modulo2-Lorena/exercicio20.swift
//
//  main.swift
//  Exercicio20
//
//  Created by Lorena Rodrigues Bruno on 24/08/2018.
//  Copyright © 2018 Lorena Rodrigues Bruno. All rights reserved.
//

import Foundation

//for linha in 1...8 {
//    for coluna in 1...8 {
//        print("*", terminator: "")
//    }
//    print("\n")
//
//}

func figura1() {
    for contador in 1...8 {
        for coluna in 1...8 {
            if coluna <= contador {
                print("*", terminator:"")
            } else {
                print(" ", terminator:"")
            }
        }
        print("\n")
    }
}

func figura2() {
    for contador in stride(from: 8, to: 0, by: -1) {
        for coluna in 1...8{
            
            if coluna <= contador {
                print("*", terminator:"")
            } else {
                print(" ", terminator:"")
            }
        }
        print("\n")
    }
}

func figura3(){
    for contador in 1...8 {
        for coluna in 1...8 {
            if coluna > contador {
                print("*", terminator:"")
            } else {
                print(" ", terminator:"")
            }
        }
        print("\n")
    }
}

func figura4() {
    for contador in stride(from: 8, to: 0, by: -1) {
        for coluna in 1...8{
            
            if coluna > contador {
                print("*", terminator:"")
            } else {
                print(" ", terminator:"")
            }
        }
        print("\n")
    }
}

figura1()
print("\n")
figura2()
print("\n")
figura3()
print("\n")
figura4()
print("\n")


